src/analysis/processing/qgsalgorithmshortestline.cpp

```diff
@@ -0,0 +1,180 @@
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "DESTINATION" ), QObject::tr( "Destination layer" ), QList<int>() << QgsProcessing::TypeVectorAnyGeometry ) );
+  addParameter( new QgsProcessingParameterEnum( QStringLiteral( "METHOD" ), QObject::tr( "Method" ), QStringList() << "distance to nearest point on geometry" << "distance to geometry centroid", false, QVariant( "boundary" ) ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "NEIGHBORS" ), QObject::tr( "Maximum number of neighbors" ), QgsProcessingParameterNumber::Integer, 1, false, 1 ) );
+  addParameter( new QgsProcessingParameterDistance( QStringLiteral( "DISTANCE" ), QObject::tr( "Maximum distance" ), QVariant(), QString( "SOURCE" ), true ) );
+  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ), QObject::tr( "Shortest lines" ) ) );
+}
+
+bool QgsShortestLineAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback * )
+{
+  mSource.reset( parameterAsSource( parameters, QStringLiteral( "SOURCE" ), context ) );
+  if ( !mSource )
+    throw QgsProcessingException( invalidSourceError( parameters, QStringLiteral( "SOURCE" ) ) );
+
+  mDestination.reset( parameterAsSource( parameters, QStringLiteral( "DESTINATION" ), context ) );
+  if ( !mSource )
+    throw QgsProcessingException( invalidSourceError( parameters, QStringLiteral( "DESTINATION" ) ) );
+
+  mMethod = parameterAsInt( parameters, QStringLiteral( "METHOD" ), context );
+
+  mKNeighbors = parameterAsInt( parameters, QStringLiteral( "NEIGHBORS" ), context );
+
+  double paramMaxDist = parameterAsDouble( parameters, QStringLiteral( "DISTANCE" ), context ); //defaults to zero if not set
+  if ( paramMaxDist )
+    mMaxDistance = paramMaxDist;
+
+  return true;
+}
+
+QVariantMap QgsShortestLineAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  if ( mKNeighbors > mDestination->featureCount() )
+    mKNeighbors = mDestination->featureCount();
+
+  QgsFields fields = QgsProcessingUtils::combineFields( mSource->fields(), mDestination->fields() );
+  fields.append( QgsField( QStringLiteral( "distance" ), QVariant::Double ) );
+
+  QString dest;
+  std::unique_ptr< QgsFeatureSink > sink( parameterAsSink( parameters, QStringLiteral( "OUTPUT" ), context, dest, fields, QgsWkbTypes::MultiLineString, mSource->sourceCrs() ) );
+  if ( !sink )
+    throw QgsProcessingException( invalidSinkError( parameters, QStringLiteral( "OUTPUT" ) ) );
+
+  QgsFeatureRequest request = QgsFeatureRequest();
+  request.setDestinationCrs( mSource->sourceCrs(), context.transformContext() );
+
+  QgsSpatialIndex idx = QgsSpatialIndex( mDestination->getFeatures( request ), nullptr, QgsSpatialIndex::FlagStoreFeatureGeometries );
```

Comment:
```suggestion
  QgsSpatialIndex idx = QgsSpatialIndex( mDestination->getFeatures( request ), feedback, QgsSpatialIndex::FlagStoreFeatureGeometries );
```

Reply:
Although it would be better to follow the approach from qgsalgorithmangletonearest.cpp:212  so that users get progress reports as the spatial index is built